script.js

```diff
@@ -0,0 +1,236 @@
+// api url
+const API_URL =
+	'https://raw.githubusercontent.com/alexsimkovich/patronage/main/api/data.json';
+
+// declaration of items creating product list
+let $ulList;
+let $listItem;
+
+// error message paragraph
+let $errorInfo;
+
+// The array which store all items from API
+let $dataArr = [];
+// The array which store all items in cart
+let $cartArr = [];
+
+// declaration of items creating cart
+let $ulListCart;
+let $cartOrder;
+let $cartEmpty;
+let $cartTotal;
+
+const main = () => {
+	prepareDOMElements();
+	prepareDOMEvents();
+};
+
+const prepareDOMElements = () => {
+	$ulList = document.querySelector('#product-container');
+	$cartOrder = document.querySelector('#cart-order');
+	$ulListCart = document.querySelector('#cart-list');
+};
+
+const prepareDOMEvents = () => {
+	$ulList.addEventListener('click', orderButtonHandler);
+	$ulListCart.addEventListener('click', removeButtonHandler);
+};
+
+const fetchData = async () => {
+	// loading spinner before we fetch all the data
+	const ulList = document.querySelector('#product-container');
+	const loadingSpinner = document.createElement('div');
+	loadingSpinner.classList.add('spinner');
+	ulList.appendChild(loadingSpinner);
+
+	const response = await fetch(API_URL);
+
+	if (!response.ok) {
+		const message = `An error has occured: ${response.status}`;
+		throw new Error(message);
+	}
+	const data = await response.json();
+
+	// loading spinner removed
+	loadingSpinner.remove();
+
+	return data;
+};
+
+fetchData()
```

Comment:
można to usprawnić, np.:
  
```
fetchData().then(res => renderPizzaList(res))

i całą logikę renderowania przenieść do osobnej funkcji
```

Reply:
Zastosowałem wszystkie podpowiedzi do fetch'a. Logike renderowania przeniosłem do osobnej funkcji dla listy i dla renderowania informacji o błędzie.